StructuralVariantAttributeList doesn't crash on null `evidence` field

phildarnowsky-broad opened the pull request:
The underlying cause of all this was some bugs in the `StructuralVariant` type. Fixing those has made it possible to fix the bug that originally brought this to our attention, and to do a lot of cleaning of this region of the code as a fringe benefit.

Fixes #1380

rileyhgrant reviewed (approved):
Nice one. 

I wasn't able to actually confirm locally, as (I think?) something about the pattern of the URL doesn't allow me to hit that in local testing? I remember Elissa encountering a problem like this.

Anyhoo, the fix in the code looks solid.

---

![Screenshot 2024-02-28 at 10 56 55](https://github.com/broadinstitute/gnomad-browser/assets/59549713/55cf5383-de27-4beb-91db-8a364e26a3e4)

phildarnowsky-broad commented:
@rileyhgrant I'm not clear on why that won't load locally. However, now that I'm trying it in a demo, there's a lot more stuff on this page that needs fixing, so I'm going to work on this branch more.

phildarnowsky-broad commented:
@rileyhgrant I have done a few little experiments here and I think the reason that you can't load this page locally is due to the `.` in the variant name. I bet Express is trying to interpret everything after the dot as a filename extension, so I'm going to see what lever I have to pull to get it to quit doing that.

rileyhgrant commented:
> @rileyhgrant I have done a few little experiments here and I think the reason that you can't load this page locally is due to the `.` in the variant name. I bet something is trying to interpret everything after the dot as a filename extension, so I'm going to see what lever I have to pull to get it to quit doing that.

Aha, yeah that makes sense. I'm curious as to what lever that is, iirc with Elissa we just ended up realizing a URL like this worked in demos, so never properly looked into why it wouldn't work locally with the v4 push and all.

phildarnowsky-broad commented:
@rileyhgrant I've added here some fixups to address a regression I introduced, as well as some changes to allow routing of our problem variant locally. Ready for re-review.

rileyhgrant reviewed (approved):
LGTM!
